woocommerce/woocommerce
# Migrate E2E Tests to Puppeteer

### Changes proposed in this Pull Request:

This PR seeks to rewrite the E2E tests replacing Selenium with Puppeteer.

In order to do so, the Babel version had to be updated.

The Puppeteer-based test environment was largely copied from Happy Tools. Now that things are working, its probably best to use a setup that more closely matches Gutenberg ([#](https://github.com/WordPress/gutenberg/tree/master/packages/e2e-tests)).

#### What is left to be done

1. Remove defunct testing dependencies - mocha, Selenium, etc
2. Fix timing issues in "new product" test - Puppeteer often moves too quickly for the AJAX and UI blocking behavior on the "add product" screen.
3. Finish WIP checkout test.
4. Rewrite `npm test:grep` and `npm test:single` to work with new env
5. Use `@wordpress/e2e-test-utils` as a dependency once `pressKeyWithModifier()` is updated. See: 
1b4dd51

### How to test the changes in this Pull Request:

1. `npm install`
2. `npm test` _(modify the test script in `package.json` to add `--dev` argument to see browser)_

#### If you're having trouble..

1. Try `rm -rf ./node_modules && rm package-lock.json && npm install`
2. Make sure the `url` property in your config ends with a trailing `/`
3. Some of the tests expect a fresh import of the `tests/e2e-tests/data/e2e-db.sql` database
